library/docs/user-guide.md

```diff
@@ -221,3 +221,49 @@ class DebugDefaultOptionFactory : DefaultOptionFactory {
   private val <T : Feature<T>> T.options get() = javaClass.options
 }
 ```
+## Parent–child relationship
+
+Feature flags can be supervised using `FeatureFlag.supervisorOption` property. Whenever supervisor has its option different from the value in this property then the supervised feature flag cannot return any other option than a default one. Option can still be set via `Laboratory` but it will not be exposed as long as a feature flag is not supervised. This relationship is recursive meaning that grandparents control grandchildren indirectly.
```

Comment:
Child defining its supervisor forces both to be in the same module. Would inverse relationship be feasible at all? Example:
```kotlin
interface Feature {
  val supervisedFeatures: Set<Class<out Feature<*>>>
    get() = emptySet()
}

enum class ChristmasTheme(
  override val supervisedFeatures: Set<Class<out Feature<*>>>
) : Feature<ChristmasTheme> {
  Enabled(supervisedFeatures = setOf(Greeting::class.java, Background::class.java)),
  Disabled(supervisedFeatures = emptySet())
}
```

Reply:
It doesn't force to be in the same module. Child module can depend on parent module. I would prefer it the other way around but unfortunately it's impossible.

Child needs to declare what it is being supervised by. It is needed because when you interact with `Laboratory` the only source of information is a feature that you pass.

Also it is safer because you can't define that feature flag is being supervised by more than one feature.

Reply:
Thanks for explaining. I have no other questions - the feature looks great.

---

>Also it is safer because you can't define that feature flag is being supervised by more than one feature.

I guess I could come up with use cases where a feature could depend on 2 parents, but that seems... unnecessarilly complicated.